Budgets/Budgets.Domain.UnitTests/BudgetCategoryShould.cs

```diff
@@ -56,7 +57,7 @@ public void BudgetCategoryShouldHaveAssignedMoneyAtMonthYear()
                 .WithMoneyAssigned(monthYear, expectedMoneyAssigned)
                 .Build();
 
-            Assert.Equal(expectedMoneyAssigned, budgetCategory.MoneyAssigned.Sum(moneyAssigned => moneyAssigned.Value.Amount));
+            budgetCategory.MoneyAssigned.Sum(moneyAssigned => moneyAssigned.Value.Amount).Should().Be((decimal)expectedMoneyAssigned);//�Assert.Equal(expectedMoneyAssigned, budgetCategory.MoneyAssigned.Sum(moneyAssigned => moneyAssigned.Value.Amount));?
```

Comment:
explicit decimal cast in expectedMoneyAssigned not required

Reply:
Previous assert was performing an implicit cast from decimal to Money. In this case the asserted member defines the type in subsequent assertations, which is decimal as the Sum operation return that type. 

Would it be better to perform a explicit cast to Money for the Sum operation result? Or look the way of ensuring a sum operation which returns Money?

Reply:
Value is Money, but Amount is decimal

![image](https://user-images.githubusercontent.com/8351573/173635214-9f4626c3-0d1c-45d7-9c22-1ff4b5056aca.png)

Reply:
In the previous version I had the explicit cast from Money, until I found Amount

Reply:
I have a solution that could work. It is about an extension of Linq.Sum so it is possible to aggregate Money types. I followed: https://stackoverflow.com/questions/4132820/can-you-overload-sum-to-add-custom-types. Working locally, we can review it together